src/session-pool.ts

```diff
@@ -667,25 +561,18 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
     // back into the pool. This will prevent the session to be marked as leaked if
     // the pool is closed while the session is being prepared.
     this._traces.delete(session.id);
-    this._pendingPrepare++;
-    session.type = types.ReadWrite;
-    this._prepareTransaction(session)
-      .catch(() => (session.type = types.ReadOnly))
-      .then(() => {
-        this._pendingPrepare--;
-        this._release(session);
-      });
+    this._prepareTransaction(session);
```

Comment:
Is this still necessary? I assume that it does not actually call the `BeginTransaction` RPC, but why do we need it for all sessions that are released into the pool here? And why do we not do it if the session is released into the pool above to unblock a waiter (on line 556)?

Reply:
No, it's not necessary any more. Transactions will be prepared(initialized) anyway during the session acquiring anyway.